include/oneapi/dpl/pstl/hetero/dpcpp/parallel_backend_sycl_merge.h

```diff
@@ -156,11 +172,22 @@ __serial_merge(const _Rng1& __rng1, const _Rng2& __rng2, _Rng3& __rng3, const _I
         // One of __rng1_idx_less_n1 and __rng2_idx_less_n2 should be true here
         // because 1) we should fill output data with elements from one of the input ranges
         // 2) we calculate __rng3_idx_end as std::min<_Index>(__rng1_size + __rng2_size, __chunk).
-        if (__rng1_idx_less_n1 && __rng2_idx_less_n2 && __comp(__rng2[__rng2_idx], __rng1[__rng1_idx]) ||
-            !__rng1_idx_less_n1)
-            __rng3[__rng3_idx] = __rng2[__rng2_idx++];
+        if constexpr (__can_use_ternary_op<_Rng1, _Rng2>().value)
```

Comment:
Please add a comment that the use of ternary operator is an optimization, and perhaps a TODO to understand why the usual if-else is slower.